src/main/java/org/testng/internal/TestInvoker.java

```diff
@@ -321,26 +322,19 @@ private String checkDependencies(ITestNGMethod testMethod) {
       invoker.invokeBeforeGroupsConfigurations(arguments);
     }
 
-    long maxTimeOut = -1; // 10 seconds
-
-    for (IWorker<ITestNGMethod> tmw : workers) {
-      long mt = tmw.getTimeOut();
-      if (mt > maxTimeOut) {
-        maxTimeOut = mt;
-      }
-    }
+    long maxTimeOut = workers.parallelStream()
+        .max(Comparator.comparingLong(IWorker::getTimeOut))
```

Comment:
`map` first, then `max`: it will avoid the double `IWorker::getTimeOut` call

Reply:
Fixed